textinput/textinput.go

```diff
@@ -91,6 +92,66 @@ func (c CursorMode) String() string {
 	}[c]
 }
 
+// ValidateBool validates boolean-like inputs only
+// i.e 'y', 'n', 't', 'f', '1', '0' (case insensitive)
+func ValidateBool(runes []rune, m Model) []rune {
+	out := make([]rune, 0)
+
+	for _, r := range runes {
+		if len(m.value) <= 0 {
+			_, parseErr := strconv.ParseBool(string(r))
+			if parseErr == nil || r == 'y' || r == 'Y' || r == 'n' || r == 'N' {
+				out = append(out, r)
+			}
+		}
+	}
+
+	return out
+}
+
+// ValidateInt validates integer-only inputs.
+func ValidateInt(runes []rune, m Model) []rune {
+	out := make([]rune, 0)
+	for _, r := range runes {
+		_, parseErr := strconv.ParseInt(string(r), 10, 32)
```

Comment:
I'd probably default to 64 bit here.

Reply:
Again, any argument why since we're parsing a single digit?